platforms/allwinner-d1/boards/src/bin/mq-pro.rs

```diff
@@ -123,3 +128,63 @@ fn main() -> ! {
 
     d1.run()
 }
+
+fn init_i2c_puppet_irq(gpio: &mut d1_pac::GPIO, plic: &mut Plic) -> &'static WaitCell {
+    use mnemos_d1_core::plic::Priority;
+    use d1_pac::Interrupt;
+
+    static I2C_PUPPET_IRQ: WaitCell = WaitCell::new();
+
+    // configure pin mappings:
+
+    // the i2c_puppet PI_INT line is on the GCLK0/GPIO4 pin on the Pi header,
+    // according to this schematic:
+    // https://github.com/sqfmi/beepy-hardware/blob/d051e65fd95fdadd83154378950b171a001125a8/KiCad/beepberry-schematic-v1.pdf
+    //
+    // according to the MangoPi MQ Pro schematic, that pin is routed to PB7 on
+    // the D1: https://mangopi.org/_media/mq-pro-sch-v12.pdf
+    gpio.pb_cfg0.modify(|_r, w| {
```

Comment:
You might want to enable internal pullups here, depending on whether the rp2040 is in open-drain or push-pull mode.

Reply:
Ah, just kidding! There are hardware pullups on the beepy board.

<img width="151" alt="Screenshot 2023-07-30 at 19 46 32" src="https://github.com/tosc-rs/mnemos/assets/2097964/6038657d-c78c-43ac-ad98-9d74eeafdf0b">

Reply:
yup, i was also looking at the schematic and was about to mention that. i'll add a comment.